### DISCUSSION TITLE
Samples-transfer-06

### DISCUSSION THREAD

**Discussion by Akshay-a-j** [eclipse-edc/Samples#96] [Q&A]
Hi, 
I am currently working on implementing [sample-transfer-06](https://github.com/eclipse-edc/Samples/blob/main/transfer/transfer-06-consumer-pull-http/README.md). I made some modifications to the code to experiment with different permission methods. Specifically, I added print statements to check the contract agreement and contract offer in [ContractNegotiationProtocolServiceImpl.java](https://github.com/eclipse-edc/Connector/blob/56bab11e455930b2b76b19ac99d093b5d76482e7/core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/service/contractnegotiation/ContractNegotiationProtocolServiceImpl.java#L46) and [ContractOffer.java](https://github.com/eclipse-edc/Connector/blob/56bab11e455930b2b76b19ac99d093b5d76482e7/spi/control-plane/contract-spi/src/main/java/org/eclipse/edc/connector/contract/spi/types/offer/ContractOffer.java#L30).

To test my modifications, I published the [Connector](https://github.com/eclipse-edc/Connector/tree/56bab11e455930b2b76b19ac99d093b5d76482e7) to MavenLocal. My expectation was that when I build [sample-transfer-06](https://github.com/eclipse-edc/Samples/blob/main/transfer/transfer-06-consumer-pull-http/README.md) again, it would fetch the updated version from the local .m2 folder.

However, when I run [sample-transfer-06](https://github.com/eclipse-edc/Samples/blob/main/transfer/transfer-06-consumer-pull-http/README.md), I encounter the following error, and the contract agreement and contract offer are not printed as expected.

"edc:stateTimestamp"
Provider end: "TransferProcess 3e2c8503-6825-4d45-a491-bef80db17ea9 is now in state STARTED"
Consumer end: "No checker found for process fc13e9b7-6a1d-46bb-ba5e-7488e07526fb. The process will not advance to the COMPLETED state."

Can you please help me on this? 

Thank you

**Comment by Akshay-a-j** [comment]
Hi @ndr-brt, could you please help me with this?

Thank you

**Reply by ndr-brt** [reply]
It seems like a build problem, not an EDC one, after publishing it on your local maven did you updated the version on the samples repository?

not strictly related... why you added print statements? you can verify everything by calling the rest endpoints. If you need some innard detail, debugging would be easier (without having to build and publish).

**Reply by Akshay-a-j** [reply]
Thank you, @ndr-brt. I have successfully updated the version in the samples repository. However, upon attempting to build transfer-06, I encountered a build failure due to the inability to locate the "edc-build/org.eclipse.edc.edc-build.gradle.plugin/0.2.1-SNAPSHOT/maven-metadata.xml" file within the ".m2/repository/org/eclipse/edc" directory. Subsequently, I attempted to rebuild the connector and move it to MavenLocal(). Unfortunately, the connector build itself is also failing, generating the following error:: 

19 tests completed, 5 failedz

> Task :extensions:common:json-ld:test FAILED

FAILURE: Build failed with an exception.

**Comment by johnnychoque** [comment]
Hi @ndr-brt,
I seem to have the same problem as @Akshay-a-j. Yesterday I downloaded the latest versions of edc (v0.2.1-1-g899948f09) and edc-sample (no release) but I get similar error messages when I try to check the transfer status:

```
DEBUG 2023-08-22T13:16:19.437687357 TransferProcess ad156d92-de7f-42e0-83ca-1263a71318b3 is now in state INITIAL
DEBUG 2023-08-22T13:16:19.438628031 Process ad156d92-de7f-42e0-83ca-1263a71318b3 is now INITIAL
DEBUG 2023-08-22T13:16:19.439492042 Transfer Process created ad156d92-de7f-42e0-83ca-1263a71318b3
DEBUG 2023-08-22T13:16:19.594961357 TransferProcess ad156d92-de7f-42e0-83ca-1263a71318b3 is now in state PROVISIONING
DEBUG 2023-08-22T13:16:19.596529137 TransferProcess: ID ad156d92-de7f-42e0-83ca-1263a71318b3. Provisioning
DEBUG 2023-08-22T13:16:19.599973394 TransferProcess ad156d92-de7f-42e0-83ca-1263a71318b3 is now in state PROVISIONED
DEBUG 2023-08-22T13:16:19.601105671 TransferProcess ad156d92-de7f-42e0-83ca-1263a71318b3 is now in state REQUESTING
DEBUG 2023-08-22T13:16:19.602823227 TransferProcess: ID ad156d92-de7f-42e0-83ca-1263a71318b3. Send TransferRequestMessage to http://localhost:19194/protocol
DEBUG 2023-08-22T13:16:19.629273341 TransferProcess: ID ad156d92-de7f-42e0-83ca-1263a71318b3. Send TransferRequestMessage to http://localhost:19194/protocol
DEBUG 2023-08-22T13:16:19.629874156 TransferProcess ad156d92-de7f-42e0-83ca-1263a71318b3 is now in state REQUESTED
DEBUG 2023-08-22T13:16:20.399317817 DSP: Incoming TransferStartMessage for transfer process: ad156d92-de7f-42e0-83ca-1263a71318b3
DEBUG 2023-08-22T13:16:20.401948064 TransferProcess ad156d92-de7f-42e0-83ca-1263a71318b3 is now in state STARTED
WARNING 2023-08-22T13:16:20.608872023 No checker found for process ad156d92-de7f-42e0-83ca-1263a71318b3. The process will not advance to the COMPLETED state.
WARNING 2023-08-22T13:17:20.626914696 No checker found for process ad156d92-de7f-42e0-83ca-1263a71318b3. The process will not advance to the COMPLETED state.
WARNING 2023-08-22T13:18:20.642221843 No checker found for process ad156d92-de7f-42e0-83ca-1263a71318b3. The process will not advance to the COMPLETED state.
WARNING 2023-08-22T13:19:20.658759919 No checker found for process ad156d92-de7f-42e0-83ca-1263a71318b3. The process will not advance to the COMPLETED state.
WARNING 2023-08-22T13:20:20.671682333 No checker found for process ad156d92-de7f-42e0-83ca-1263a71318b3. The process will not advance to the COMPLETED state.

```
In my case, I understand that I have used the updated versions of the repositories but the problem keeps happening, can you give us any idea how to solve it?

Thanks in advance.

**Reply by xdxxxdx** [reply]
I have the same situation here.

**Reply by ndr-brt** [reply]
This is not blocking the transfer, it's just telling that the transfer will not be `COMPLETED` automatically because there's no checker registered. It doesn't affect the sample functionality, but eventually a status checker could be added for sampling purposes. Would you mind create an issue for it?

**Comment by johnnychoque** [comment]
One remark. Although the transfer fails to go to COMPLETED status, the backend service logs show the json with the information to read the data. Executing a query using "authCode" will get them.
